static/js/lightbox.js

```diff
@@ -122,13 +124,20 @@ exports.open = function (image, options) {
                 $source = $image.attr("src");
             }
         }
-        var message = message_store.get($message.attr("zid"));
-        if (message === undefined) {
-            blueslip.error("Lightbox for unknown message " + $message.attr("zid"));
+        var user_payload;
```

Comment:
Let's call this `sender_full_name`, since that's what it is.

Reply:
Also, I think this PR would be a good opportunity for you to split these changes into 2 commits.  This PR is simple enough it's not strictly necessary, but the ideal structure would be:
* First commit: Extract arguments to `payload` below are variables `sender_full_name` (and the existing `$parent`).
* Second commit: Add the support for using the lightbox from the compose preview.

Reply:
@timabbott  I couldn't understand the meaning of the suggested first commit ? Please elaborate a bit more.
Also I strongly feel that .open() is doing more stuff than it should. Should I try to seperate out the logic of detecting video and preview out of the .open method and move it in the caller area? Should I do this in this PR request ? And what is ideal commit structure for this proposed change?

Reply:
Ahh, the point of 2 commits is to separate out the non-functionality-changing refactor that moves code around from the functional change.  As I said, with this PR it's basically unnecessary, but as you start working on larger projects, it's really valuable to have the discipline of breaking up commits in this way.  Why?  Because reviewing a clean refactoring commit that promises to have no functional effect is generally really quick, and then the remaining changes of adding the new if/else statement that would go in the second commit would be really clean and easy to verify as well.

So, the end Git tree after this PR would be the same with 2 commits as with 1 -- the commits would just provide a more reviewable story explaining your changes and why they are correct.

You can ask for help from others in #development help on chat.zulip.org if need be; anyone in the community should be able to help explain how to do this.

(Check out our [GitHub guide](https://zulip.readthedocs.io/en/latest/git-guide.html) and especially [commit guidelines](https://zulip.readthedocs.io/en/latest/contributing/version-control.html) for more details as well.)